src/pseudoaligner.rs

```diff
@@ -85,7 +85,7 @@ impl<K: Kmer + Sync + Send> Pseudoaligner<K> {
         if kmer_pos >= left_extend_threshold && node_id.is_some() {
             let mut last_pos = kmer_pos - 1;
             let mut prev_node_id = node_id.unwrap();
-            let mut prev_kmer_offset = kmer_offset.unwrap() - 1;
+            let mut prev_kmer_offset = if kmer_offset.unwrap() > 0 { kmer_offset.unwrap() - 1 } else { 0 };
```

Comment:
I ran into cases while testing where kmer_offset = 0 and this caused underflow because it's unsigned.